core/server/master/src/main/java/alluxio/master/file/meta/MountTable.java

```diff
@@ -454,6 +510,49 @@ public Journaled getDelegate() {
     return mState;
   }
 
+  /**
+   * ValidatedPathPair is used to store the [alluxioPath: ufsPath] pair that has been validated as
+   * a legal MountTable Entry. ValidatedPathPair can only be constructed and returned by
+   * {@link MountTable#validateMountPoint(AlluxioURI, AlluxioURI)}.
+   */
+  public static final class ValidatedPathPair implements Closeable {
```

Comment:
Per our discussion, this object serves 4 purposes:
1. Holds an Alluxio path
2. Holds a UFS path
3. Holds a LockResource
4. Releases the lock on `close()`

If we move the lock/unlock outside the object and do that manually, this object can be replaced by a `Pair<AlluxioPath, UfsPath>`. This will make the code a bit more expressive.